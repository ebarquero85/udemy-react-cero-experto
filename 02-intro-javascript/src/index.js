
const apiKey = 'J3SWjyhj4sWgiQfeffpZY5yr8O6CoMU8';

const peticion = fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apiKey}`);

peticion
    .then( response => response.json() )
    .then( ({ data }) => {
        
        console.log(data);
        const { url } = data.user.avatar_url;
        
        console.log(url);
        const img = document.createElement('img');
        
        img.src = url;
        document.body.append( img );

    });
