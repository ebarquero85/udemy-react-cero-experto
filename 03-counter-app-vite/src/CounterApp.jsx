import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);

  const handleAdd = (e) => setCounter( (c) => c + 1 );

  const handleReset = (e) => setCounter( () => 0 );

  const handleSubstract = (e) => setCounter( (c) => c - 1 );

  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>
      <button onClick={handleAdd}> +1 </button>
      <button aria-label="btn-reset" onClick={handleReset}> Reset </button>
      <button onClick={handleSubstract}> -1 </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};
