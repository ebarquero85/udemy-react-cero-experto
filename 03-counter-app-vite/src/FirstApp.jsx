import PropTypes from "prop-types";

const FirstApp = ({ title, subTitle, name}) => {
  return (
    <>
      <h1 data-testid="test-title" >{ title }</h1>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: 'Edgard Barquero',
  subTitle: 'No hay subtitulo',
  title: 'No hay t√≠tulo',
}

export default FirstApp;
