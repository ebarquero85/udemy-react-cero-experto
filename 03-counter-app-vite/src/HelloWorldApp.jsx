

const HelloWorldApp = () => {
  return (
    <>
        HelloWorldApp
    </>
  )
}

export default HelloWorldApp