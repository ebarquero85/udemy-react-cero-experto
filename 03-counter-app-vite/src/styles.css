html, body {
    background-color: black;
    color: beige;
    padding: 70px;
}

button {
    background-color: beige;
    color: black;
    padding: 10px;
    border: none;
    border-radius: 5px;
    font-size: 20px;
    font-weight: bold;
    cursor: pointer;
    margin-right: 10px;
}