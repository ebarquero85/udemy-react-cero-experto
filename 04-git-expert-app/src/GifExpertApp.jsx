import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  
  const [categories, setCategories] = useState([]);

  const addCategories = (value) => {
  
    if(categories.includes(value)) return;
    
    setCategories( [value, ...categories] ); //Operador spreed.
    //setCategories( cat => ['BARQUERO', ...cat]);

  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={ addCategories }/>

      {
        categories.map( category => (
          <GifGrid 
            key={ category } 
            category={ category } 
          />
        ))
      }
    </>
  );

};