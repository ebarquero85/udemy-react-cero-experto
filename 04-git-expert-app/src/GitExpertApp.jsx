import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GitExpertApp = () => {
  
  const [categories, setCategories] = useState([]);

  const addCategories = (value) => {
  
    if(categories.includes(value)) return;
    
    setCategories( [...categories, value] ); //Operador spreed.
    //setCategories( cat => ['BARQUERO', ...cat]);

  };

  return (
    <>
      <h1>GitExpertApp</h1>

      <AddCategory 
        onNewCategory={ addCategories }
      />

      {
        categories.map( category => (
          <GifGrid 
            key={ category } 
            category={ category } 
          />
        ))
      }

    </>
  );

};