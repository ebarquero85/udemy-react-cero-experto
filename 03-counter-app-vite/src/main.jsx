import React from 'react';
import ReactDOM from 'react-dom/client';
import HelloWorldApp from './HelloWorldApp';
import FirstApp from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HelloWorldApp />
    <FirstApp name="Edgard" age={37} />
  </React.StrictMode>
);