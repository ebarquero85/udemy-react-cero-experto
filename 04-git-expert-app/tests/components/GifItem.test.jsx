import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components'

describe('Pruebas en <GifItem />', () => {

    const title = 'Dog Driving';
    const url = 'https://media4.giphy.com/media/kiBcwEXegBTACmVOnE/giphy-downsized-medium.gif?cid=b76bcd7c04ukm3m4lpf5402vfwuh1voqbg2t44molis3iyrx&rid=giphy-downsized-medium.gif'
    test('Debe de hacer match con el snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url }/> );
        expect( container ).toMatchSnapshot();
        
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
    
        render( <GifItem title={ title } url={ url }/> );
        //screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    })

    test('Debe de mostrar el tÃ­tulo en el componente', () => { 
    
        render( <GifItem title={ title } url={ url }/> );
        expect(screen.getByText(title)).toBeTruthy();
    
    })
    
});