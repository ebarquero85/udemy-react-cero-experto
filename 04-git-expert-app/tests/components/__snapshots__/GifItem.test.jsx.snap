// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dog Driving"
      src="https://media4.giphy.com/media/kiBcwEXegBTACmVOnE/giphy-downsized-medium.gif?cid=b76bcd7c04ukm3m4lpf5402vfwuh1voqbg2t44molis3iyrx&rid=giphy-downsized-medium.gif"
    />
    <p>
      Dog Driving
    </p>
  </div>
</div>
`;
